### ISSUE TITLE
Problems finding words through underscores

### ISSUE THREAD

**Issue by marinaaaniram** [meilisearch/meilisearch#3996]
`meilisearch engine v1.1.1`

```
{
    "displayedAttributes": [
        "*"
    ],
    "searchableAttributes": [
        "table_name_short",
        "table_name",
        "column_name"
    ],
    "filterableAttributes": [
        "key_type",
        "tags"
    ],
    "sortableAttributes": [
        "key",
        "key_type",
        "popularity"
    ],
    "rankingRules": [
        "sort",
        "words",
        "typo",
        "proximity",
        "attribute",
        "exactness",
        "release_date:asc",
        "rank:desc"
    ],
    "stopWords": [],
    "synonyms": {},
    "distinctAttribute": null,
    "typoTolerance": {
        "enabled": true,
        "minWordSizeForTypos": {
            "oneTypo": 5,
            "twoTypos": 9
        },
        "disableOnWords": [],
        "disableOnAttributes": []
    },
    "faceting": {
        "maxValuesPerFacet": 100
    },
    "pagination": {
        "maxTotalHits": 1000
    }
}
```

I have the following entries in the index:

```
 table_name = 'crm.action_log'
 table_name = 'analytics.stat_crm_aggregated_data'
 table_name = 'crm.api_token'
 table_name = 'analytics.stat_crm_data'
 table_name = 'crm.other'
```

Then I go to meilisearch web, typing `crm` and see:

```
 table_name = 'crm.action_log_action_history'
 table_name = 'crm.api_token'
 table_name = 'crm.other'
```

What's wrong with the underscore and dots? 
And how can I use them? 
In technical content, words are often separated by either underscores or dots, not spaces.

**Comment by marinaaaniram** [comment]
Hey! Any comments?

**Comment by jiangbo212** [comment]
meilisearch searches by prefix, and underscores are not token separators, so for this search of yours, the results are correct.

**Comment by marinaaaniram** [comment]
How can I set to meilisearch treat underscores as spaces? Is it technically possible?

**Comment by Kerollmops** [comment]
Hey @marinaaaniram 👋

Thank you for your detailed report! We just released [a candidate version of Meilisearch v1.4.0](https://github.com/meilisearch/meilisearch/releases/tag/v1.4.0-rc.0), which lets you change the default separator and define your owns.

Please try it and give us feedback 🥕

**Comment by curquiza** [comment]
I close this issue due to a lack of answers. Nothing is definitive of course, feel free to ask to re-open if needed 😊
